src/js/components/FormGroupComponent.jsx

```diff
@@ -2,17 +2,31 @@ var React = require("react/addons");
 
 var FormGroupComponent = React.createClass({
   displayName: "FormGroupComponent",
+
   propTypes: {
     attribute: React.PropTypes.string,
     children: React.PropTypes.object.isRequired,
     errors: React.PropTypes.array,
     help: React.PropTypes.string,
     label: React.PropTypes.string,
-    model: React.PropTypes.object.isRequired
+    model: React.PropTypes.object.isRequired,
+    validator: React.PropTypes.object.isRequired
+  },
+
+  getInitialState: function () {
+    return {
+      validationError: null
+    };
   },
 
   onInputChange: function (event) {
-    this.props.model.set(event.target.name, event.target.value);
+    var props = this.props;
+
+    props.model[event.target.name] = event.target.value;
```

Comment:
Well, the "correct" way would be to always let the parent component modify its own state as we actually need these to be in sync. When the state is updated, in the parent, the child will receive the new value of the prop automatically. I'm inclined to say `NewAppModalComponent` should pass `FormGroupComponent` a callback to be fired inside `onInputChange`, which will create a deep copy of the model and the assign it to the state via the `this.setState()`. We should probably use [React immutability helpers](https://facebook.github.io/react/docs/update.html) for this.